cli/pkg/timescaledb_secrets/create.go

```diff
@@ -0,0 +1,344 @@
+		b[i] = passwordChars[b[i] % uint8(len(passwordChars))]
+	}
+	return b, nil
+}
+
+func (t *TSDBSecretsInfo) createTimescaleDBCredentials() error {
+	secretName := t.ReleaseName + "-credentials"
+	exists, err := k8s.CheckSecretExists(secretName, t.Namespace)
+	if err != nil {
+		return err
+	}
+
+	if exists {
+		fmt.Printf("using existing %s secret\n", secretName)
+		return nil
+	}
+
+	fmt.Printf("Creating TimescaleDB %s secret\n", secretName)
+	repPass, err := randomPassword(10)
+	if err != nil {
+		return err
+	}
+	adminPass, err := randomPassword(10)
+	if err != nil {
+		return err
+	}
+	superUserPass, err := randomPassword(10)
+	if err != nil {
+		return err
+	}
+	sec := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretName,
+			Namespace: t.Namespace,
+		},
+		Data: map[string][]byte{
+			"PATRONI_REPLICATION_PASSWORD": repPass,
+			"PATRONI_admin_PASSWORD":       adminPass,
+			"PATRONI_SUPERUSER_PASSWORD":   superUserPass,
+		},
+		Type: "Opaque",
+	}
+
+	return k8s.CreateSecret(sec)
+}
+
+func generateCerts() ([]byte, []byte, error) {
+	priv, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to generate rsa key: %v", err)
+	}
+
+	serialNumberLimit := new(big.Int).Lsh(big.NewInt(1), 128)
+	serialNumber, err := rand.Int(rand.Reader, serialNumberLimit)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to generate serial number: %v", err)
+	}
+
+	template := x509.Certificate{
+		SerialNumber:          serialNumber,
+		Subject: pkix.Name{
+			CommonName: "Tobs TimescaleDB Fake Certificate",
+			Organization: []string{"Acme Co"},
+		},
+		NotBefore:             time.Now(),
+		NotAfter:              time.Now().Add(time.Hour * 24 * 1000),
+		KeyUsage:              x509.KeyUsageDigitalSignature,
+		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth},
+		BasicConstraintsValid: true,
+	}
+
+	derBytes, err := x509.CreateCertificate(rand.Reader, &template, &template, &priv.PublicKey, priv)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to create certificate: %v", err)
+	}
+
+	certOut, err := pemEncodeCert(derBytes)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to write data to cert.pem: %v", err)
+	}
+
+	keyOut, err := pemEncodePrivateKey(priv)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to encode private key: %v", err)
+	}
+
+	return certOut, keyOut, nil
+}
+
+type s3Details struct {
+	BucketName     string
+	EndpointName   string
+	EndpointRegion string
+	Key            string
+	Secret         string
+}
+
+func (t *TSDBSecretsInfo) createTimescaleDBCertificates() error {
+	fmt.Print(`
+WARNING: Using a generated self-signed certificate for TLS access to TimescaleDB.
+         This should only be used for development and demonstration purposes.
+         To use a signed certificate, use the "--tls-timescaledb-cert" and "--tls-timescaledb-key"
+         flags when issuing the tobs install command.
```

Comment:
I would personally add a check after this message, something like `Do you wish to proceed? [y/n]`. 

I haven't seen this behavior anywhere in TOBS though, I would be fine to add it in a later PR if we decide to do it at all.

Reply:
So basically we avoided that because we want one cmd to install tobs without any user inputs. To make it as easy as possible.

Reply:
I think this is ok for now. It's a pretty fat warning